### ISSUE TITLE
ProxyRoomSource not working

### ISSUE THREAD

**Issue by pteich** [StreamMachine/StreamMachine#33]
I've tried to connect StreamMachine to an existing Icecast server using your ProxyRoomSource as input (fallback config setting). But it doesn't worked with the version in your master branch. 

It works with some minor changes that you can review here:
[compare changes] (https://github.com/StreamMachine/StreamMachine/compare/master...pteich:master)

But I'm not sure if I'm missing something (your code tries to create a new parser with an non existing method _new_parser and I don't know why).

**Comment by ewr** [comment]
Welcome to some ancient code! We haven't run the proxy source in production for a couple years now (basically since we added native source support), and what you see is the result...

`_new_parser` used to live in the Base source, but Base now takes care of parsers automatically. Your changes were exactly in the right direction, but I went ahead and cleaned up a few other things I saw in there.

Let me know if those changes work for you.
